sentinel-one/tasks/main.yml

```diff
@@ -0,0 +1,36 @@
+---
+# Use ephemeral AWS credentials provided in the CI/CD actions. Make sure the
+# following variables are set:
+#   tmp_aws_access_key
+#   tmp_aws_secret_key
+#   tmp_aws_session_token
+#   sentinelone_debian_installer
+- name: Download SentinelOne installer from private S3 bucket
+  ansible.builin.shell: "aws --no-progress s3 cp s3://{{ sentinelone_bucket }}/sentinelone/SentinelAgent_linux_v22_1_1_4.deb /tmp/SentinelAgent_linux_v22_1_1_4.deb"
+  args:
+    creates: /tmp/SentinelAgent_linux_v22_1_1_4.deb
+  environment:
+    AWS_ACCESS_KEY_ID: "{{ tmp_aws_access_key }}"
+    AWS_SECRET_ACCESS_KEY: "{{ tmp_aws_secret_key }}"
+    AWS_SESSION_TOKEN: "{{ tmp_aws_session_token }}"
+  when: ansible_distribution == 'Ubuntu' or ansible_distribution == 'Debian'
+
+- name: Install SentinelOne package
+  ansible.builtin.apt:
```

Comment:
Probably want to add `become: yes` to anything that will need root. Some playbooks are run as root at the playbook level, and some aren't, so ensuring everything works when not run as root to begin with will work more reliably

Reply:
Good idea, will get that later tonight

Reply:
Added `become: yes` to all of the steps.  The file download to /tmp/ technically doesn't need it, but I don't see a reason not to just so we can avoid permissions problems depending on how /tmp/ is setup.